x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -45,7 +45,6 @@ const registerFeatures = (features: SetupPlugins['features']) => {
     app: [PLUGIN_ID, 'kibana'],
     catalogue: [PLUGIN_ID],
     order: 2300,
-    excludeFromBasePrivileges: true,
```

Comment:
This was an intentional design decision when this feature was introduced in https://github.com/elastic/kibana/pull/106669 (cc @patrykkopycinski). I don't remember the full context, but I believe the team wanted to ensure that osquery was explicitly granted to end users, rather than "rolling up" into the preconfigured all/read base privileges.

Altering this behavior has historically been considered a **breaking change**, and as such cannot be made in a minor release.

Reply:
> I think that the teams consensus (in here: https://github.com/elastic/kibana/issues/128768) is that we want to change this.

Ok, great!


> IF we decide to still go with the change, how do you suggest going forward with this ? Are you suggesting waiting for post-v8 version?

Yes, my recommendation is to wait for the next major version. Otherwise we will grant access to osquery to users who previously didn't have access, and this may have been done intentionally by an administrator. Granting this additional access simply by upgrading their stack constitutes a "surprise on upgrade", and when it comes to authorization/access controls, that's not a good thing.

Reply:
Hi @legrego, apologies for the slow reply on this one. The team wanted to discuss it as a group before getting back to you, and I was out of office for a bit.

We’d like to move forward with this during the 8.x release period, for a couple reasons. 1) After a customer raised this issue, it became clear that the current setting can be experienced as a bug. If an administrator is setting up a user role that has All Kibana access, it would make sense that would apply to Osquery as well. 2) It’s not clear yet what the plans are for 9.x. So if possible, we’d like to fix this sooner rather than later.

While surprising people on upgrade is not ideal, in this case, we would plan to clearly document this change so that people know what to expect. And for full Osquery access, Administrators would still need to take additional steps so that people can use it: to view osquery data, users need `Read` access to the `logs-osquery_manager.result*` index, and to use Osquery itself, the Osquery Manager integration must be deployed to Elastic Agents. So while users with `All` may get access to a new nav item after this change, they wouldn't be able to see any sensitive data without fulfilling the other Osquery requirements.

If it would be easier to discuss this one live, I’m happy to set up a meeting, just let me know. Thanks!

Reply:
Hey @melissaburpo, thanks for taking the time to explain your reasoning, and the mitigating factors that are in place to prevent unintended usage of this feature. That makes me feel quite a bit better, but I would like to discuss with @jportner before we proceed. He's on PTO for the next few days, but I'll try to follow up with him towards the end of the week.

Reply:
Hi folks, I spoke with Larry about this --
I agree it's not ideal but it's OK to make this particular change.

My one suggestion is to relabel this issue with `release_note:fix` (instead of `release_note:skip`) so that we can capture this change in our release notes.